### PR TITLE
Fix Data Layer capabilities

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Fix Data Layer capabilities

#### WHY

There was a bug in naming. horologist_wear vs horologist_watch.

Now prefer horologist_phone and horologist_watch.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: datalayer-phone/src/res/values/wear.xml [hunk 1]
```diff
@@ -3,5 +3,7 @@
     <string-array name="android_wear_capabilities">
         <!-- Used to indicate that the app is installed on this device -->
         <item>data_layer_app_helper_device_phone</item>
+        <!-- Used to indicate the device is a phone -->
```
Comment (luizgrp) [inline datalayer-phone/src/res/values/wear.xml:6]: is the one above in use? same for `data_layer_app_helper_device_watch`

File: auth-sample-wear/src/main/res/values/wear.xml [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-     limitations under the License.
--->
-
-<resources xmlns:tools="http://schemas.android.com/tools">
-    <string-array
-        name="android_wear_capabilities"
-        translatable="false"
-        tools:ignore="UnusedResources">
-        <!-- declaring the provided capabilities -->
-        <item>horologist_wear</item>
```
Comment (luizgrp) [inline auth-sample-wear/src/main/res/values/wear.xml:23]: this capability is also in `sample/src/main/res/value/wear.xml`

**Review by luizgrp — APPROVED** [review]

File: datalayer-phone/src/res/values/wear.xml (hunk 1 above)
Comment (yschimke) [reply datalayer-phone/src/res/values/wear.xml:6]: Around here https://github.com/google/horologist/blob/4ba21e728db803308ce4967b0048ba99e3ba6153/datalayer/src/main/java/com/google/android/horologist/data/DataLayerAppHelper.kt#L157

File: auth-data-phone/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -19,7 +19,5 @@
     xmlns:tools="http://schemas.android.com/tools"
     package="com.google.android.horologist.auth.data.phone">
 
-    <uses-feature android:name="android.hardware.type.watch" />
```
Comment (yschimke) [inline auth-data-phone/src/main/AndroidManifest.xml:22]: @luizgrp this ok?

Comment (luizgrp) [reply auth-data-phone/src/main/AndroidManifest.xml:22]: yes, thanks for that!

**Pull request merged**